DEVDOCS-4484: [update] Stencil, custom templates now support front matter (#1652)
bigcommerce/dev-docs: docs/stencil-docs/storefront-customization/custom-templates.md

## Specifying custom front matter

> When adding front matter directly to a custom template, you must add front matter to the default template when using Stencil CLI. However, when applying a theme to the storefront, it is possible to specify front matter directly to the custom template. See [Using Front Matter](/stencil-docs/storefront-customization/using-front-matter) for more information on using front matter. 

Reviewer: ```suggestion
You can add front matter directly to a custom template. See [Using Front Matter](/stencil-docs/storefront-customization/using-front-matter) for more information on using front matter. 
```

> The following example shows data you can inject into `templates/pages/brand.html` to access data in `templates/pages/custom/brand/custom-brand.html`. After [mapping](/stencil-docs/storefront-customization/custom-templates#mapping-multiple-urls) the custom template to the default page, the cart object displays on the page.

Reviewer: It looks like you can completely delete line 123 as no injections to default template required.

> The following example shows data you can inject into `templates/pages/brand.html` to access data in `templates/pages/custom/brand/custom-brand.html`. After [mapping](/stencil-docs/storefront-customization/custom-templates#mapping-multiple-urls) the custom template to the default page, the cart object displays on the page.

Reviewer: @bc-traciporter I believe we can remove lines 121 -123 above. Stencil-cli versions `5.2.1+` supports front matter on custom templates. So if a customer defines a custom template with a front matter in his local theme (in `config.stencil.json`), this front matter will be taken into account by stencil-cli. So you don't need to change default template/front matter as we use directly what we have in custom template.

<!-- type: tab-end -->

<!-- theme: warning -->
> #### Stencil CLI versus storefront

> > When adding front matter directly to a custom template, Stencil CLI behaves as explained above. You must add front matter to the default template. However, when applying a theme to the storefront, it is possible to specify front matter directly to the custom template.

Reviewer: Stencil CLI starting from version `5.2.1` has the same behaviour as a regular mode (eg. live store) - both refular and CLI modes use front matter from custom template, not default one.
